ArduSub/script_button.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#if AP_SCRIPTING_ENABLED
+
+#include <AP_Common/AP_Common.h>
+
+// Joystick button object for use in Lua scripts.
+//
+// Provide 2 ways to use a joystick button:
+//      is_pressed() returns true if the button is currently (as of the most recent MANUAL_CONTROL msg) pressed
+//      get_and_clear_count() returns the number of times the button was pressed since the last call
+//
+class ScriptButton {
+public:
+    ScriptButton(): pressed(false), count(0) {}
+
+    void press();
+
+    void release();
+
+    [[nodiscard]] bool is_pressed() const;
```

Comment:
```suggestion
    bool is_pressed() const WARN_IF_UNUSED;
```
We have a macro for this that we use else where.

Reply:
Thanks, I will update.